xmake/modules/private/action/require/impl/package.lua

```diff
@@ -995,7 +995,7 @@ function _compatible_with_previous_librarydeps(package, opt)
     -- has been checked?
     local compatible_checked = package:data("librarydeps.compatible_checked")
```

Comment:
ohmm. I've forgotten why I judged it in the first place.

I can't find where it was set. librarydeps.compatible_checked is supposed to use caching to speed up compatibility checking.

package:data is a private data interface and should not be set by the user.

Reply:
I don't think it's doing anything, at least for now, and it should be removed.

Reply:
I'm using it at the end of a on_fetch callback.
```lua

    on_fetch(function (package, opt)
        local nazara = os.getenv("NAZARA_ENGINE_PATH")
        if not nazara or not os.isdir(nazara) then
            return
        end

-- ...

        package:data_set("librarydeps.compatible_checked", true)

        return fetchInfo
    end)
```

because otherwise, xmake wants to install the package even though it found it because its dependencies hash don't match (which is not a problem here). We already discussed this here: https://github.com/xmake-io/xmake/issues/2719 and I ran into this issue again, with no solution. I though setting `"librarydeps.compatible_checked"` would help.

Reply:
There is no mention of using package:data("librarydeps.compatible_checked") to bypass it. As I recall, there was no good discussion of a solution.

Also, package:data("librarydeps.compatible_checked") is a private interface, and calling it directly by the user would be very hacky and difficult to maintain later. As it is, I've forgotten what it does.

Reply:
It looks like it was introduced here: https://github.com/xmake-io/xmake/commit/2a8af8db1d8b6e05f2a751a3320a1c45596ac0e5 with no purpose (probably a leftover of another implementation).

So yeah, I think it should be removed. However I still have no way to disable strict dependency checking with on_fetch, which makes `on_fetch` useless when you use strict dependencies.

Reply:
The issue is that my dependent packages use `set_policy("package.strict_compatibility", true)` as well

Reply:
you can modify here

```diff
    local strict_compatibility = project.policy("package.librarydeps.strict_compatibility")
    if strict_compatibility == nil then
        strict_compatibility = package:policy("package.librarydeps.strict_compatibility")
    end
+    if strict_compatibility == false then
+       return true
+    end
```

Reply:
I removed it and add this check. you can try it. https://github.com/xmake-io/xmake/commit/34eb729eb862e7415028c84970f3b2c830d95cd9

and you can add new pr to improve it.